cmd/install/install.go

```diff
@@ -0,0 +1,233 @@
+	ErrCurlMissing    = errors.New("please install curl and set it in path")
+	kubectlPath       = "/usr/local/bin/kubectl"
+	argocdPath        = "/usr/local/bin/argocd"
+	Yellow            = color.New(color.FgHiYellow).SprintFunc()
+	Green             = color.New(color.FgGreen).SprintFunc()
+	Red               = color.New(color.FgRed).SprintFunc()
+)
+
+func NewCommand() *cobra.Command {
+	command := &cobra.Command{
+		Use:               "install",
+		Short:             "Install required tools for Arlon",
+		Long:              "Install kubectl, Argocd cli, check git cli",
+		DisableAutoGenTag: true,
+		Example:           "arlon install",
+		RunE: func(c *cobra.Command, args []string) error {
+			var err error
+			// Install kubectl and point it to the kubeconfig
+			_, err = installKubectl()
+			if err == ErrKubectlPresent {
+				fmt.Println(Green("✓") + " kubectl is already present in the path")
+			} else if err != nil {
+				fmt.Println(Red("x ")+"Error while installing kubectl ", err)
+			} else {
+				fmt.Println(Green("✓") + " Successfully installed kubectl")
+			}
+
+			fmt.Println()
+			_, err = verifyGit()
+			if err == ErrGitPresent {
+				fmt.Println("git is already present in the path")
+			} else {
+				fmt.Println(Yellow("! ") + "Install git cli")
+			}
+
+			fmt.Println()
+			_, err = installArgoCD()
+			if err == ErrArgoCDPresent {
+				fmt.Println(Green("✓") + " argocd is already present in the path")
+			} else if err != nil {
+				fmt.Println(Red("x ")+"Error while installing argocd ", err)
+			} else {
+				fmt.Println(Green("✓") + " Successfully installed argocd")
+			}
+
+			return nil
+		},
+	}
+	return command
+}
+
+// Check if kubectl is installed and if not then install kubectl
+func installKubectl() (bool, error) {
+	_, err := exec.LookPath("kubectl")
+	if err != nil {
+		fmt.Println(Yellow("! ") + "kubectl is not installed")
+		errInstallKubectl := installKubectlPlatform()
+		if errInstallKubectl != nil {
+			return false, ErrKubectlFail
+		} else {
+			return true, nil
+		}
+	}
+	return true, ErrKubectlPresent
+}
+
+// Check if git is installed and if not, then prompt user
+func verifyGit() (bool, error) {
+	_, err := exec.LookPath("git")
+	if err != nil {
+		return false, err
+	}
+	return true, ErrGitPresent
+}
+
+// Check if argocd is installed and if not, then install argocd
+func installArgoCD() (bool, error) {
+	_, err := exec.LookPath("argocd")
+	if err != nil {
+		fmt.Println(Yellow("! ") + "argocd cli is not installed")
+		errInstallArgocd := installArgoCDPlatform()
+		if errInstallArgocd != nil {
+			fmt.Println(" → Error installing argocd")
+			return false, ErrArgoCDFail
+		} else {
+			return true, nil
+		}
+
+	}
+	return true, ErrArgoCDPresent
+}
+
+// Check the platform and on the basis of that install kubectl
+func installKubectlPlatform() error {
+	var err error
+	osPlatform := runtime.GOOS
+	fmt.Println(" → Installing kubectl")
+	switch osPlatform {
+	case "darwin":
```

Comment:
logic for darwin and linux look identical and could be combined

Reply:
Done